test/functional/cursor.test.js

```diff
@@ -4089,5 +4089,40 @@ describe('Cursor', function () {
         beta: 1
       });
     });
+
+    it('should use allowDiskUse option on sort', {
+      metadata: { requires: { mongodb: '>=4.4' } },
+      test: withMonitoredClient('find', function (client, events, done) {
+        const db = client.db('test');
+        db.collection('test_sort_allow_disk_use', (err, collection) => {
+          expect(err).to.not.exist;
+          const cursor = collection.find({}).sort(['alpha', 1]).allowDiskUse();
+          cursor.next(err => {
+            expect(err).to.not.exist;
+            const { command } = events.shift();
+            expect(command.sort).to.deep.equal({ alpha: 1 });
+            expect(command.allowDiskUse).to.be.true;
+            cursor.close(done);
+          });
+        });
+      })
+    });
+
+    it('should error if allowDiskUse option used without sort', {
+      metadata: { requires: { mongodb: '>=4.4' } },
+      test: withClient(function (client, done) {
+        const db = client.db('test');
+        db.collection('test_sort_allow_disk_use', (err, collection) => {
+          expect(err).to.not.exist;
+          try {
+            collection.find({}).allowDiskUse();
```

Comment:
```
expect(() => collection.find({}).allowDiskUse()).to.throw(/allowDiskUse must follow sort/)
```
also nit: maybe the error message could be a bit more informative, "allowDiskUse requires a sort specification" or something. You could write this code `collection.find({}, { allowDiskUse: true })` and it wouldn't be clear because `allowDiskUse` isn't technically following anything (this language seems specific to the builder pattern)

Reply:
> this language seems specific to the builder pattern

That was actually my intent, I was trying to respect the language in the docs I quoted in my other comment where it says `.allowDiskUse()` must come after a `.find().sort()`.

Now that I think a bit more about it, that could be confusing, especially if users are mixing using options and the builder pattern, e.g. `.find({}, { sort: ... }).allowDiskUse().`. 

Maybe it would be better to not throw an error at all?

Reply:
I think it makes sense to tell the user that their `allowDiskUse` specification will have no effect if some sort isn't specified, I just wouldn't make it specific to needing to follow an `allowDiskUse` builder method. But that's my two cents.